src/crimson/os/seastore/seastore_types.cc

```diff
@@ -129,6 +129,81 @@ std::ostream &operator<<(std::ostream &out, const paddr_t &rhs)
   return out << ">";
 }
 
+journal_seq_t journal_seq_t::add_offset(
+      journal_type_t type,
+      seastore_off_t off,
+      seastore_off_t roll_start,
+      seastore_off_t roll_size) const
+{
+  assert(offset.is_absolute());
+  assert(off != MIN_SEG_OFF);
+  assert(roll_start >= 0);
+  assert(roll_size > 0);
+
+  segment_seq_t jseq = segment_seq;
+  seastore_off_t joff;
+  if (type == journal_type_t::SEGMENTED) {
+    joff = offset.as_seg_paddr().get_segment_off();
+  } else {
+    assert(type == journal_type_t::CIRCULAR);
+    auto boff = offset.as_blk_paddr().get_block_off();
+    assert(boff <= MAX_SEG_OFF);
```

Comment:
BLOCK_OFF_MAX ?

Reply:
get_block_off() can return the maximum of 56 bit value. On the other hand, MAX_SEG_OFF is int32_t. Therefore, this assert can be triggered even though the blk_paddr_t has correct offset. Am I wrong?

Reply:
The specific assert is necessary to detect the potential overflow with the current `seastore_off_t`.

See https://github.com/ceph/ceph/pull/47698#discussion_r951039417, I think this is a separate issue. Before that, I'll assert that the related configurations are in the reasonable ranges.

For example https://github.com/ceph/ceph/blob/d9526dc2a380fd91c3316eac1906cd97ad244a64/src/crimson/os/seastore/async_cleaner.h#L568-L570

Is it reasonable to you?

Reply:
OK